core/router/gateway_status.go

```diff
@@ -21,11 +20,5 @@ func (r *router) HandleGatewayStatus(gatewayID string, status *pb_gateway.Status
 		}
 	}()
 
-	gateway := r.getGateway(gatewayID)
-	gateway.LastSeen = time.Now()
-	err = gateway.Status.Update(status)
-	if err != nil {
-		return err
-	}
-	return nil
+	return r.getGateway(gatewayID).HandleStatus(status)
```

Comment:
I thought I have addressed that in your previous comment, but it seems that it disappeared.
Please consider https://play.golang.org/p/wMSQ3XbT17 , which I mentioned in my previous reply
Otherwise, I do not exactly understand what the issue is here